app/src/main/java/com/example/next_show/adapters/ShowAdapter.java

```diff
@@ -23,22 +24,23 @@
 import com.zerobranch.layout.SwipeLayout;
 
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
 
 public class ShowAdapter extends RecyclerView.Adapter<ShowAdapter.ViewHolder> {
     // instance variables
     private Context context;
     private List<Show> currentShows; // currently displayed shows on recycler view
-    private List<Show> allShows;     // list of all trending or recommended shows retrieved from Trakt
+    private List<Show> allShows; // all shows from Trakt call
+    private ShowFilter showFilter;
     private NavigationInterface nav;
 
-    public ShowAdapter (Context context, List<Show> shows, NavigationInterface nav){
+    public ShowAdapter (Context context, List<Show> shows, NavigationInterface nav, ShowFilter showFilter){
         this.context = context;
         this.currentShows = shows;
         this.nav = nav;
+        this.showFilter = showFilter;
 
-        allShows = shows;
+        showFilter.setShowsList(shows);
```

Comment:
Yeah, there's no real logic that you have in here.  Just set the shows like you're doing here in the outer fragment.  Then the adapter doesn't need to know any information about this filtering API